portal/server/types/db-models/users.ts

```diff
@@ -0,0 +1,28 @@
+import { WithId } from 'mongodb';
+import { PortalRole } from '../portal/portal-role';
+import { Bank } from './banks';
+import { UnixTimestamp } from '../date';
+import { Prettify } from '../types-helper';
+
+export type PortalUser = Prettify<
+  WithId<{
```

Comment:
I'd be tempted to just have `_id` as string in the front end, as thats all we care about really.

@toddym42 do you have an opinion on this?

Reply:
Would mean we dont need mongodb in a frontend service, which really smells to me

Reply:
I think it's technically correct to have `_id: ObjectId` here as this is meant to represent the db-model, not the API response. It doesn't feel quite right to have the DB model here though, as it's not an object that is every actually present in the portal ui project. I think we're defining it here because the `PortalSessionUser` is derived from it, but we lose the benefit of deriving from the actual DB model because this is just a copy and there's no guarantee that this is kept in line with the version in `dtfs-central-api`.

I think then maybe it just makes sense to remove this and explicitly declare all the properties of `PortalSessionUser`. After the workspaces work has been merged to main (hopefully soon!) we could share the DB models in the common library and then have rethink.

In `PortalSessionUser` I would have the `_id` as `string` because it's been serialised at that point when returned by the API.

Reply:
`_id` will be a string in the frontend as `ObjectId` is serialised when sent as JSON, so `WithId` should be replaced with `_id: string` :)

Reply:
I think you've summarised my thoughts a lot more coherently than me Mike :D